src/point-cloud-octree.ts

```diff
@@ -35,9 +35,15 @@ import { byLevelAndIndex } from './utils/utils';
 export interface PickParams {
   pickWindowSize: number;
   pickOutsideClipRegion: boolean;
+
+  /**
+   * In the case of custom pixel position coordinates, this property
+   * will be used for the pick window position.
+   */
+  pixelPos: Vector3;
 }
 
-interface IPickState {
+export interface IPickState {
```

Comment:
Why should this be exported? It is an internal detail of the picking system and shouldn't be used from the outside.

Reply:
In the current configuration is it impossible to introduce custom changes to the pick material and the pick render target. For instance, some properties for a material are not considered in the `updatePickMaterial()`. The same with the render target.

Reply:
Sure, but why would you want to modify those? The pick material should have only the bare minimum to be able to perform picking. The fact that we are using a material and render-target at all is just an implementation detail. We could just as easily use the octree and do the picking in the CPU and the client shouldn't have to care about that.

Maybe if you provide an example of the exact thing you are trying to accomplish, then we could come up with some option/param to make it possible through the public interface of the octree.

Reply:
In the case of multiple viewports you have to be able to correctly set a current viewport size for the pick render target and scissors. In the case of a material there is for example a property `useFilterByNormal` which reduces the number of visible points. Moreover, when someone introduces custom features for shaders, you need to reflect those changes in the pick material as well.

Reply:
So how about having an optional callback in the pick params that would be called with the material and render target after or in `updatePickMaterial()`?